os/sys/process.h

```diff
@@ -318,11 +334,12 @@ struct process {
 #define PROCESS_NAME_STRING(process) ""
 #else
   const char *name;
-#define PROCESS_NAME_STRING(process) (process)->name
+#define PROCESS_NAME_STRING(process) ((process) != NULL ? (process)->name : "")
 #endif
   PT_THREAD((* thread)(struct pt *, process_event_t, process_data_t));
   struct pt pt;
-  unsigned char state, needspoll;
+  uint8_t state;
+  bool needspoll;
```

Comment:
bool value on ARM platform takes 4byte. therefore new structure takes more RAM

Reply:
I checked on the Contiki-NG ARM-based platforms and size of bool is 1 byte.